Source/WebCore/dom/Element.cpp

```diff
@@ -946,7 +946,7 @@ void Element::setHovered(bool value, Style::InvalidationScope invalidationScope,
 
     if (auto* style = renderStyle(); style && style->hasEffectiveAppearance()) {
         CheckedPtr renderer = this->renderer();
-        renderer->theme().stateChanged(*renderer, ControlStyle::State::Hovered);
+        renderer->theme().conditionalRepaint(*renderer, ControlStyle::State::Hovered);
```

Comment:
Can we just inline the condition? I don't think we need a method at all.

Reply:
We only call `repaint()` on Adwaita currently (all other themes return false for `supportsHover()`). But I see now that `supportsHover()` is a public method, so yeah, we could do that.